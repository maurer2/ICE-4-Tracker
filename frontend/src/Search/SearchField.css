.input {
  margin: 0;
	border: 1px solid gray;
	padding: 0.5rem;
	display: inline-block;
	vertical-align: middle;
	white-space: normal;
	background: #fff;
	line-height: 1;
}
