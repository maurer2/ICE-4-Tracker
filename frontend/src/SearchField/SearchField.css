.input {
	margin: 0;
	padding: 0.5rem;
	display: block;
	border: 1px solid gray;
	vertical-align: middle;
	white-space: normal;
	background: #fff;
	line-height: 1;
}

.search {
	width: 100%;
}

