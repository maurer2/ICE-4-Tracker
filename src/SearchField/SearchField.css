.input {
	margin: 0;
	padding: 0.5rem;
	display: block;
	border: 1px solid gray;
	white-space: normal;
	background: #fff;
	line-height: 1;
}

.search {
	width: 100%;
}

