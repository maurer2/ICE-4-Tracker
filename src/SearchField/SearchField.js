import React, { Component } from 'react';
import classNames from 'classnames';

import styles from './SearchField.module.css';

class SearchField extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: '',
      hasFocus: false,
    };

    this.handleChangeEvent = this.handleChangeEvent.bind(this);
    this.handleFocusEvent = this.handleFocusEvent.bind(this);
    this.handleBlurEvent = this.handleBlurEvent.bind(this);
    this.handleKeyUpEvent = this.handleKeyUpEvent.bind(this);
  }

  handleChangeEvent(event) {
    this.setState({
      value: event.target.value,
    });
  }

  handleFocusEvent(event) {
    this.setState({
      hasFocus: true,
    }, () => {
      this.props.cbFieldFocus(this.state.hasFocus);
    })
  }

  handleBlurEvent(event) {
    this.setState({
      hasFocus: false,
    }, () => {
      this.props.cbFieldFocus(this.state.hasFocus);
    })
  }

  handleKeyUpEvent(event) {
    const key = event.key;

    if (key === undefined) {
      return;
    }

    this.props.cbKeyboardEvent(key);
  }

  render() {
    const { value } = this.state;

    return (
      <input
        className={ classNames(styles.input, styles.search) }
        type="search"
        name="search"
        placeholder="Zugnummer eingeben"
        autoComplete="off"
        value={ value }
        onChange={ this.handleChangeEvent }
        onFocus={ this.handleFocusEvent }
        onBlur={ this.handleBlurEvent }
        onKeyUp={ this.handleKeyUpEvent }
      />
    )
  }
}

export default SearchField;
