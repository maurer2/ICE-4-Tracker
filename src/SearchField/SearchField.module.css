.field {
  margin: 0;
  padding: 0.75rem;
  display: block;
  border: 1px solid gray;
  white-space: normal;
  background: #fff;
  line-height: 1;
  font-size: 1rem;
}

.field,
.field:active,
.field:hover {
  outline: none;
}

.search {
  width: 100%;
}
