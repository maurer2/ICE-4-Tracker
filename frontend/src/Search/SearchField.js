import React, { Component } from 'react';
import './SearchField.css';

class SearchField extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ''
    };
  }

  handleChangeEvent(event) {
    this.setState({
      value: event.target.value,
    });
  }

  render() {
    return (
      <input type="search" className="input" name="search" placeholder="Zugnummer eingeben"
        value={ this.state.value } onChange={ this.handleChangeEvent.bind(this) } />
    )
  }
}

export default SearchField;
