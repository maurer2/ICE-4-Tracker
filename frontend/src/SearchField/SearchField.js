import React, { Component } from 'react';
import './SearchField.css';

class SearchField extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: '',
      trainNumbers: ['500', '600', '700'],
    };

    this.handleChangeEvent = this.handleChangeEvent.bind(this);
  }

  handleChangeEvent(event) {
    this.setState({
      value: event.target.value,
    });
  }

  handleKeyUpEvent(event) {
    let key = event.key;

    if (key === undefined) {
      key = String.fromCharCode(event.keyCode);
    }
  }

  render() {
    return (
      <div className="form-row">
        <input type="search" className="input search" name="search" placeholder="Zugnummer eingeben" autoComplete="off"
          value={ this.state.value } onChange={ this.handleChangeEvent } />
        <ul className={`suggestions ${(this.state.value !== '') ? 'suggestions--are-visible' : '' }`}>
          { this.state.trainNumbers
            .map((trainNumber, index) => <li className="suggestion" key={ index }>{ trainNumber }</li>) }
        </ul>
      </div>
    )
  }
}

export default SearchField;
