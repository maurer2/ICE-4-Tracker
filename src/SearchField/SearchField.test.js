import React from 'react';
import SearchField from './SearchField';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe('SearchField', () => {
  const mockedHandleFieldFocus = jest.fn();
  const mockedHandleKeyboardEvent = jest.fn();

  const wrapper = shallow(
    <SearchField
      showLoader={false}
      trainNumbers={['500', '555']}
      cbFieldFocus={mockedHandleFieldFocus}
      cbKeyboardEvent={mockedHandleKeyboardEvent}
    />
  );

  it('SearchField should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
