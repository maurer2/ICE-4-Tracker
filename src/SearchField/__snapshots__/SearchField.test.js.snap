// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField SearchField should match snapshot 1`] = `
<input
  autoComplete="off"
  className="input search"
  name="search"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyUp={[Function]}
  placeholder="Zugnummer eingeben"
  type="search"
  value=""
/>
`;
